---
layout: default
title:  'Single Cell Genomics Tutorial'
---

## Overview of steps in this exercise

We will have a lunch break. We will split the discussion of results into two parts, one when you have the assembly results and then another one to summarize the whole day. 
Here is a schematic [Workflow](slides/scg_workflow.pdf) of what you will be doing.  

1. [Connecting to UPPMAX](tutorials/connectToUppmax.md)  
2. [Familiarizing with data](tutorials/scg_part2.md)  
3. [Single-cell genome assembly](tutorials/scg_part3.md)  
3.1. [Organize working folder](tutorials/scg_part3_1.md)  
3.2. [Pre-processing](tutorials/scg_part3_2.md)  
3.3. [Assembly](tutorials/scg_part3_3.md)  
3.4. [Assessing assembly quality using Quast](tutorials/scg_part3_4.md)  
3.5. [Gene prediction using Prodigal](tutorials/scg_part3_5.md)  
3.6. [Running completeness estimates](tutorials/scg_part3_5.md)  
3.7. [Identifying ribosomal RNAs](tutorials/scg_part3_7.md)  
4. [Assessing read coverage and chimera checking](tutorials/scg_part4.md)  
4.1 [Reads mapping](tutorials/scg_part4_1.md)  
4.2 [Assessing coverage bias](tutorials/scg_part4_2.md)  
4.3 [Detection and inspection of chimeric reads](tutorials/scg_part4_3.md)  
4.4 [Insert size] (tutorials/scg_part4_4.md)  
5. [Exploring your single cell genome](tutorials/scg_part5.md)  
5.1 [Contamination analysis in MEGAN](tutorials/scg_part5_1.md)  
5.2 [Functional analysis in MEGAN](tutorials/scg_part5_2.md)  
6. [Analysis of a novel single-cell genome](tutorials/scg_part6.md) 

